### ISSUE TITLE
timestamp format issue

### ISSUE THREAD

**Issue by wenwenchenbosch** [eclipse/kuksa.val#210]
https://github.com/GENIVI/iot-event-analytics/issues/105#issuecomment-867438435 reported that in kuksa.val the timestamp format is `YYYY-MM-DDTHH:mm:ss.SSSZXXX`, whereas XXX is some numeric value and which leads to problems when parsing that value.

**Comment by SebastianSchildt** [comment]
ok, expected format is `YYYY-MM-DDTHH:MM:SS.ssssssZ`    (ending after "Z" whioch is the UTC timezone designator) , https://www.w3.org/TR/NOTE-datetime 

need to check

**Comment by SebastianSchildt** [comment]
Quick test showed no problem. But maybe some memory corruption when the fractional part is smaller?

```json
Test Client> authorize ~/kuksa.val/kuksa_certificates/jwt/super-admin.json.token {
    "TTL": 1767225599, 
    "action": "authorize", 
    "requestId": "916e837f-10b8-437b-8110-9eb76b72ea63", 
    "timestamp": "2021-06-24T09:22:42.1624522962Z"
}

Test Client> setValue Vehicle.Speed 200
{
    "action": "set", 
    "requestId": "2f00a6e8-ed9b-497a-80ae-7f150609f9e9", 
    "timestamp": "2021-06-24T09:22:58.1624522978Z"
}

Test Client> getValue Vehicle.Speed 
{
    "action": "get", 
    "path": "Vehicle.Speed", 
    "requestId": "dd5922e1-8d31-4f69-86c2-00bd535d1ed5", 
    "timestamp": "2021-06-24T09:22:58.1624522978Z", 
    "value": "200"
}
```

**Comment by SebastianSchildt** [comment]
Or a problem with  the alpine image... Need to check....

**Comment by llakie** [comment]
The numeric value is gone in the meantime. The whole adapter was broken, since we expected actually a Unix-Timestamp in Milliseconds. So we compared the given number with a threshold. If the threshold was succeeded (Which the given ISO timestamp literally does) we appended the Milliseconds to it --> Since we added something to the given string, it seemed to us, that Kuksa.Val added the stuff at the end. But in the end it was our adapter. One question still remains: Did you actually sent a Unix-Timestamp and changed it to be an ISO timestamp?
